aws/services/kinesis/resource.ftl

```diff
@@ -45,6 +84,149 @@
+                    /]
+                [/#if]
+
+                [#-- Handle processor functions --]
+                [#if processorId?has_content]
+                    [#local streamProcessorArn = getArn(processorId)]
+                [/#if]
+
+                [#local bucket = destinationResources["bucket"]]
+
+                [#local rolePolicies += [
+                    getPolicyDocument(
+                        s3KinesesStreamPermission(bucket.Id) +
+                        s3AllPermission(bucket.Name, bucketPrefix) +
+                        isEncrypted?then(
+                            s3EncryptionKinesisPermission(
+                                cmkKeyId,
+                                bucket.Name,
+                                bucketPrefix,
+                                region
+                            ),
+                            []
+                        ) +
+                        processorId?has_content?then(
+                            lambdaKinesisPermission(streamProcessorArn),
+                            []
+                        ),
+                        "firehose"
+                    )
+                ]]
+
+                [#local cwLoggingConfiguration = getFirehoseStreamLoggingConfiguration(false)]
+                [#if logGroupName?has_content || logStreamName?has_content]
+                    [#local cwLoggingConfiguration = getFirehoseStreamLoggingConfiguration(true, logGroupName, logStreamName)]
+                [/#if]
+
+                [#local streamDestinationConfiguration += 
+                    getFirehoseStreamS3Destination(
+                        bucket.Id,
+                        bucketPrefix,
+                        errorPrefix,
+                        bufferInterval,
+                        bufferSize,
+                        role.Id,
+                        isEncrypted,
+                        cmkKeyId,
+                        cwLoggingConfiguration,
+                        false,
+                        {},
+                        processorId?has_content?then([
+                            getFirehoseStreamLambdaProcessor(
+                                streamProcessorArn,
+                                role.Id,
+                                bufferInterval,
+                                bufferSize
+                            )
+                        ],
+                        [])
+                    )]
+
+                [#break]
+
+            [#default]
+                [@fatal
+                    message="Invalid stream destination."
+                    detail="Supported Destinations - S3"
+                    context=destinationLink
+                /]
+                [#break]
+
+        [/#switch]
+
+        [#-- [@createKinesisDataStream 
+            id=dataStream.Id
+            name=dataStream.Name
+            keyId=dataStreamEncryptionKey
+            dependencies=dependencies
+        /] --]
```

Comment:
I don't think we should include kinesis streams as part of this process. This macro is to create a firehose not a data stream